test/DynamoCoreTests/SearchModelHostIntegrationTests.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Dynamo.Search;
+using Dynamo.Search.SearchElements;
+using Dynamo.Utilities;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+
+namespace Dynamo.Tests
+{
+    [TestFixture]
+    internal class SearchModelHostIntegrationTests : DynamoModelTestBase
+    {
+
+        private static NodeSearchModel search;
+
+        [SetUp]
+        public void Init()
+        {
+            search = new NodeSearchModel();
+        }
+
+
+        /// <summary>
+        /// Validate that the "SetParameterByName" node is found in the top 5 results from the indexed Revit 2025 nodes
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void ValidateRevitSetParameterNodeInSearch()
+        {
+            string expectedNode = "SetParameterByName";
+            string[] searchTerms = { "set parameter", "element set parameter", "setparameter", "element.setparameter" };
+            const int nodeResultsToTake = 5;
```

Comment:
This is good but I would like to be more granular, can we record the relative rank somehow?

Reply:
@QilongTang the tests were updated so we can check the expected node in a particular position in the results.
see the commit: 956e3e50eaeee58ab0369007440c9b0c14790032